docs/src/design/many_differentials.md

```diff
@@ -0,0 +1,188 @@
+
+One thing to understand about differentials is they have to form a [vector space](https://en.wikipedia.org/wiki/Vector_space)  (or something very like them).
+They need to support addition to each other, they need a zero which doesn't change what it is added to, and they need to support scalar multiplication (this isn't really required, but it is handy for things like gradient descent).
+Beyond being a vector space, differentials need to be able to be added to a primal value to get back another primal value.
+Or roughly equivalently a differential is a difference between two primal values.
+
+One thing to note in this example is that the primal does not have to be a vector space.
+We can however always transform it to be a vector space via choosing an origin, and expressing each primal value as a differential typed distance from that origin.
+As an example, consider `DateTime`. A `DateTime` is not a vector space: there is no `zero(::DateTime)`, and `DateTime`s cannot be added to each other. The corresponding differential type is any subtype of `Period`, such as `Millisecond`, `Hour`, `Day` etc.
+
+!!! info "Torsors"
+    For a related concept, see [Torsors](http://math.ucr.edu/home/baez/torsors.html).
+    
+## Natural differential
+
+For a given primal type, we say a natural differential type is one which people would intuitively think of as representing the difference between two primal values.
+It tends to already exist outside of the context of AD.
+So `Millisecond`, `Hour`, `Day` etc. are examples of _natural differentials_ for the `DateTime` primal.
+
+Note here that we already have a one primal type to many differential types relationship.
+We have `Millisecond` and `Hour` and `Day` all being valid differential types for `DateTime`.
+In this case we _could_ convert them all to a single differential type, such as `Nanoseconds`, but that is not always a reasonable decision: we may run in to overflow, or lots of allocations if we need to use a `BigInt` to represent the number of `Nanosecond` since the start of the universe.
+For types with more complex semantics, such as array types, these considerations are much more important.
+
+Natural differential types are the types people tend to think in, and thus the type they tend to write custom sensitivity rules in.
+An important special case of natural differentials is when the primal type is a vector space (e.g. `Real`,`AbstractMatrix`) in which case it is _common_ for the natural differential type to be the same as the primal type.
+One exception to this is `getindex`.
+The ideal choice of differential type for `getindex` on a dense array would be some type of spare array, due to the fact the derivative will have only one non-zero element.
+This actually further brings us to a weirdness of differential types not actually being closed under addition, as it would be ideal for the spare array to become a dense array if summed over all elements.
```

Comment:
```suggestion
This actually further brings us to a weirdness of differential types not actually being closed under addition, as it would be ideal for the sparse array to become a dense array if summed over all elements.
```